### DISCUSSION TITLE
求大神指导一下，应该怎么写，用户申请入群事件执行

### DISCUSSION THREAD

**Discussion by yixianbaowang** [MR-XieXuan/MiraiTravel#12] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/125167113/219848701-8e3bb24c-60d0-4cda-b1b1-6301717924f5.png)
![image](https://user-images.githubusercontent.com/125167113/219848794-664abb15-d177-4573-9ca6-c1942e29ab5d.png)

下面的是那个对呢？还是全部不对呢？
![image](https://user-images.githubusercontent.com/125167113/219848786-983e87cd-0557-49d6-a05e-30ffdff13e2d.png)

**Comment by MR-XieXuan** [comment]
reply_message 只能回复好友消息,群消息等。加好友只能用同意或者拒绝方式回复。

**Comment by MR-XieXuan** [comment, marked as answer]
然后你webhook收到的应该是 用户入群申请（Bot需要有管理员权限）这个事件,参考文档
https://docs.mirai.mamoe.net/mirai-api-http/api/EventType.html#%E7%94%A8%E6%88%B7%E5%85%A5%E7%BE%A4%E7%94%B3%E8%AF%B7-bot%E9%9C%80%E8%A6%81%E6%9C%89%E7%AE%A1%E7%90%86%E5%91%98%E6%9D%83%E9%99%90
 
![image](https://user-images.githubusercontent.com/95032548/219922069-596e3559-95ee-47ad-8d76-71ef2a3b573e.png)
根据webhook组件的使用教程,当接到 type 为 MemberJoinRequestEvent 时。会调用脚本的 webhook_member_join_request_event 函数 ,并且把收到的消息原封不动的传入该函数的第一个参数。所以你应该在脚本中定义的方法为 : ``webhook_member_join_request_event ($e)`` ;

**Comment by MR-XieXuan** [comment]
因为easyMirai组件更新了,可以简单的实现。
下面是一个简单的例子 :
```php
function webhook_member_join_request_event( $webhookMessage ){
      // 某个需要处理的 QQ 群
      $groupId = 604568448 ;
      // 允许入群的QQ号列表
      $accMembers = [ 3325629928 , 1438723268 ];
      // 黑名单
      $blackMembers = [ 111 , 222 , 666 ];
      
      // 如果调用函数的群是要处理的群
      if( $webhookMessage['groupId'] === $groupId ){
          if( in_array( $webhookMessage['fromId'] , $accMembers ) ){
              // 同意申请
              $this->reply_event(true);
          } else if( in_array( $webhookMessage['fromId'] , $blackMembers ) ){
              // 拒绝申请
              $this->_qqBot->reply_event(false);
          } else {
              // 不处理
              
          }
      }
  }
```
